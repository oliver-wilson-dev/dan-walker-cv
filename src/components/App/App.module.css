.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  justify-content: space-between;
}

a {
  text-decoration: none;
  color: rgb(21, 64, 204);
}

p {
  font-size: 14px;
}
.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
