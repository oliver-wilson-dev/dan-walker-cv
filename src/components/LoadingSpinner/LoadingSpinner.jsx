/*
  eslint-disable
  import/no-unresolved,
  jsx-a11y/click-events-have-key-events,
  jsx-a11y/no-static-element-interactions
*/
import React, { useEffect } from 'react';
import cn from 'classnames';

import styles from './LoadingSpinner.module.css';
import Page from '../Page';
import scrollToTop from '../../helpers/scrollToTop';

const onClick = (e) => {
  e.stopPropagation();
};

const LoadingSpinner = ({ show }) => {
  useEffect(() => {
    if (show) {
      scrollToTop();
    }
  }, [show]);

  return (
    <>
      <Page className={cn({
        [styles.loadingPage]: show
      })}
      />
      <div
        className={cn(styles.wrapper,
          {
            [styles.wrapperShow]: show,
            [styles.wrapperHide]: !show
          })}
        onClick={onClick}
      />
      <div className={cn(styles.loadingSpinner,
        {
          [styles.loadingSpinnerShow]: show,
          [styles.loadingSpinnerHide]: !show
        })}
      />
    </>
  );
};

export default LoadingSpinner;
