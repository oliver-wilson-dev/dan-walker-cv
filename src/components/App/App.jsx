import React, { Suspense, useState } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

import Navigation from '../Navigation';
import Footer from '../Footer';
import LoadingSpinner from '../LoadingSpinner';

import styles from './App.module.css';
import Fallback from '../Fallback';
import CookieDisclaimer from '../../containers/CookieDisclaimer';
import routes from '../../routes';

const { home, about, contact } = routes;

const App = () => {
  const [loading, setLoading] = useState(false);

  return (
    <div className={styles.App}>
      <Router>
        <Navigation />
        <LoadingSpinner show={loading} />
        <Suspense fallback={<Fallback setLoading={setLoading} />}>
          <Switch>
            <Route exact path={home.route}>
              <home.component />
            </Route>
            <Route path={about.route}>
              <about.component />
            </Route>
            <Route path={contact.route}>
              <contact.component />
            </Route>
          </Switch>
        </Suspense>
        <CookieDisclaimer />
        <Footer />
      </Router>
    </div>
  );
};

export default App;
