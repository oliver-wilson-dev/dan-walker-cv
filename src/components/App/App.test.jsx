import React from 'react';
import { shallow } from 'enzyme';
import LoadingSpinner from '../LoadingSpinner';
import App from './App';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: ({ children }) => <>{children}</>,
  Route: ({ children }) => <>{children}</>,
  Switch: ({ children }) => <>{children}</>
}));

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  Suspense: ({ children, fallback }) => (
    <>
      {fallback()}
      {children}
    </>
  ),
}));

jest.mock('../Navigation', () => {
  const Navigation = () => null;

  return Navigation;
});

jest.mock('../Fallback', () => {
  const Fallback = () => null;

  return Fallback;
});

jest.mock('../LoadingSpinner', () => {
  const LoadingSpinner = () => null;

  return LoadingSpinner;
});

jest.mock('../HomePage', () => {
  const HomePage = () => null;

  return HomePage;
});

jest.mock('../AboutPage', () => {
  const AboutPage = () => null;

  return AboutPage;
});

jest.mock('../ContactPage', () => {
  const ContactPage = () => null;

  return ContactPage;
});

jest.mock('../Footer', () => {
  const Footer = () => <footer />;

  return Footer;
});

jest.mock('../../helpers/lazyLoadComponent', () => ({ importFn }) => {
  importFn();
  const LazyLoadedComponent = () => null;

  return LazyLoadedComponent;
});

const render = () => shallow(<App />);

describe('App', () => {
  it('renders correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should call the LoadingSpinner with the show prop set to false', () => {
    const wrapper = render();
    expect(wrapper.find(LoadingSpinner).props()).toEqual({ show: false });
  });
});
