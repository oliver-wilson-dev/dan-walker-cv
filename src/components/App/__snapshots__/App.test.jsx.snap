// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Navigation />
    <LoadingSpinner
      show={false}
    />
    <Suspense
      fallback={
        <Fallback
          setLoading={[Function]}
        />
      }
    >
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <LazyLoadedComponent />
        </Route>
        <Route
          path="/about"
        >
          <LazyLoadedComponent />
        </Route>
        <Route
          path="/contact"
        >
          <LazyLoadedComponent />
        </Route>
      </Switch>
    </Suspense>
    <Footer />
  </BrowserRouter>
</div>
`;
